internal/server/server.go

```diff
@@ -416,8 +413,42 @@ func (s *Server) handleHealthCheck() http.HandlerFunc {
 				"err":        err,
 			}).Error(errPrefix + status)
 
-			writeJsonResponse(w, http.StatusBadGateway, healthCheckResponse{Status: status})
-			return
+			// Code below is for debugging the /healthz hanging issue
+			// Once issue is resolved, this block can be deleted
+			// Debug Start
+
+			// try another service instead of OIDC provider
+			resp, err := s.client.Get(s.debugURL.String())
+			if err == nil {
+				log.Info(errPrefix + fmt.Sprintf("s.client.Get worked for: %v", s.debugURL))
+				resp.Body.Close()
+			} else {
+				log.Info(errPrefix + fmt.Sprintf("s.client.Get also failed for: %v", s.debugURL))
+			}
+
+			// retry with a new HTTP Client
+			if err := s.initHttpClient(); err != nil {
+				log.Error(errPrefix + fmt.Sprintf("s.initHttpClient() failed: %v", err))
+			}
+
+			oidcResp, err = s.client.Get(s.issuerURL.String() + "/healthz")
+			if err == nil {
+				log.Info(errPrefix + fmt.Sprintf("on retry with new http.Client, can connect to %v", s.issuerURL))
```

Comment:
If we are successful on 2nd attempt with new HTTP Client, then we just continue on to L453.
I can monitor Splunk logs for "cannot connect to", to see if this was the fix. Or if we can set up an alert.

Reply:
I like adding the second attempt.   You can create an automated splunk alert to trigger on the log pattern.